### PR TITLE
Document the weather state functions

### PR DESCRIPTION
**Pull request by Phlosioneer**
All weather types are now documented (including WEATHER_15)

### CODE REVIEW

File: src/field_weather_effect.c [hunk 1]
```diff
@@ -2449,7 +2450,7 @@ static u8 TranslateWeatherNum(u8 weather)
     case WEATHER_DROUGHT:    return WEATHER_DROUGHT;
     case WEATHER_RAIN_HEAVY: return WEATHER_RAIN_HEAVY;
     case WEATHER_BUBBLES:    return WEATHER_BUBBLES;
-    case WEATHER_15:         return WEATHER_15;
+    case WEATHER_ALTERNATING:         return WEATHER_ALTERNATING;
```
Comment (nullableVoidPtr) [inline src/field_weather_effect.c]: Realign these switch cases

File: src/field_weather_effect.c [hunk 2]
```diff
@@ -2368,51 +2369,51 @@ void SetSav1WeatherFromCurrMapHeader(void)
...
+        if (FuncIsActiveTask(Task_DoAlternatingWeather))
+            DestroyTask(FindTaskIdByFunc(Task_DoAlternatingWeather));
+        gCurrentAlternatingWeather = WEATHER_RAIN_HEAVY;
     }
-    ChangeWeather(weather);
+    SetNextWeather(weather);
 }
 
-void sub_80AEE84(void)
+void DoPausedWeather(void)
```
Comment (huderlem) [inline src/field_weather_effect.c]: Would a better name be `ResumePausedWeather()`?

**Comment by huderlem** [comment]
Address both comments, and fix merge conflict, then this should be good for merge.

**Comment by Phlosioneer** [comment]
Updated this PR.

**Pull request merged**